src/ptrace_utils.c

```diff
@@ -55,6 +56,8 @@
 #include "common.h"
 #include "db_process.h"
 
+bool attached=false;
```

Comment:
A bool seems poorly aligned with ptrace_attach taking a pid. Maybe only attach should take a pid and should fail if a process is currently attached?

Reply:
Ah good point. The bool is used to avoid showing what appears to be an error if the sigterm or other terminating signal is received. If the -v option is used, Chericat spends most of time displaying db data on the console, which is typically when users might want to stop it if it's taking too long. During this time ptrace would not be attached to the target process and therefore the call to PT_DETACH would fail.

I had assumed that the pid would not change throughout chericat runtime. But when we support child processes tracing, this will not stand any more. Therefore I think we do need pid to be passed to ptrace_detach for it to know the right pid to detach.

Perhaps I could maintain a list of attached pids instead? Add the pid to the list when ptrace_attach is successful, and remove the pid when the ptrace_detach is successful?

Reply:
fixed.